app/controllers/automated_tests_controller.rb

```diff
@@ -114,8 +114,8 @@ def populate_autotest_manager
 
     test_specs = autotest_settings_for(assignment)
     assignment_data = assignment.assignment_properties.attributes.slice(*required_params.map(&:to_s))
-    assignment_data['token_start_date'] ||= Time.current
-    assignment_data['token_start_date'] = assignment_data['token_start_date'].iso8601
+    assignment_data['token_period'] = assignment_data['token_period'].to_i
+    assignment_data['token_start_date'] = assignment_data['token_start_date']&.change(usec: 0)&.iso8601
```

Comment:
This code is a bit strange (the `.change(usec: 0)` specifically). I don't think this is necessary, you can keep the `||= Time.current` from before